### PR TITLE
[Refactor] Rename ICEBERG_CATALOG to iceberg.catalog.type

### PR DESCRIPTION
**Pull request by mxdzs0612**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [ ] enhancement
- [x] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

Rename the property ICEBERG_CATALOG from starrocks.catalog-type to iceberg.catalog.type

This refactor will take effects on current iceberg users. The document and daily test need to be modified, too.

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by imay** [comment]
@mxdzs0612 should be compatible with original users, better to make a alias and encourage users to use the new one.
we can delete the old one later.

File: fe/fe-core/src/main/java/com/starrocks/catalog/IcebergResource.java [hunk 1]
```diff
@@ -60,7 +62,11 @@ protected void setProperties(Map<String, String> properties) throws DdlException
 
         catalogType = properties.get(ICEBERG_CATALOG);
         if (StringUtils.isBlank(catalogType)) {
-            throw new DdlException(ICEBERG_CATALOG + " must be set in properties");
+            catalogType = properties.get(ICEBERG_CATALOG_LEGACY);
+            if (StringUtils.isBlank(catalogType)) {
+                throw new DdlException(ICEBERG_CATALOG + " must be set in properties");
+            }
+            LOG.warn(ICEBERG_CATALOG_LEGACY + " will be deprecated, using " + ICEBERG_CATALOG + " instead.");
```
Comment (imay) [inline fe/fe-core/src/main/java/com/starrocks/catalog/IcebergResource.java:69]: This can be removed because users can't see this log.

**Review by imay — COMMENTED** [review]
Could you change the docs directly?

File: fe/fe-core/src/main/java/com/starrocks/catalog/IcebergResource.java (hunk 1 above)
Comment (mxdzs0612) [reply fe/fe-core/src/main/java/com/starrocks/catalog/IcebergResource.java]: Okay, I'll also revise the Chinese version of docs after this PR is merged.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 1 / 5 (20.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/IcebergResource.java|0|3|00.00%|[65, 66, 67]|
|:large_blue_circle:|com/starrocks/connector/iceberg/IcebergConnector.java|1|2|50.00%|[33]|

**Review by imay — APPROVED** [review]

**Comment by imay** [comment]
run starrocks_admit_test

**Review by stephen-shelby — APPROVED** [review]
LGTM

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Pull request merged**